rqbit: 7.0.1 -> 8.0.0

ToasterUwU opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ToasterUwU commented:
@Daru-san Only mildly related, but best place to do this i think. 
Wanna give the desktop app thing another try? Since there has been an update, maybe something works differently now and the issue is fixed or at least easier to fix.

ToasterUwU commented on pkgs/by-name/rq/rqbit/package.nix:
```diff
@@ -68,8 +68,6 @@ rustPlatform.buildRustPackage {
 
   doCheck = false;
 
-  passthru.webui = rqbit-webui;
```
it was not, not sure how that would even happened since i only pulled from master, updated the things i wanted and commited that.. I will fix that right away!
Thanks for catching that @gdamjan

gdamjan commented:
also you need to update `hash` and `npmDepsHash`

ToasterUwU commented:
I do? When i changed the version and tried to built, only the hashes i changed were not identical anymore. And since it uses the version number from the top to fetch all the things it needs, i dont see why or even to what i would need to change those hashes.

gdamjan commented:
> I do? When i changed the version and tried to built, only the hashes i changed were not identical anymore. And since it uses the version number from the top to fetch all the things it needs, i dont see why or even to what i would need to change those hashes.

yes you do, if you don't change the src hash, nix will download the old cached (7.0.1) version from the nix caches

Daru-san commented:
@ToasterUwU  I am free to help get the desktop app working. Thankfully tauri was updated to 2.0 in this release, so it _may_ work better this time.
Although we may or may not have to open a separate PR to do that, since that could take any magnitude of time to complete.

ToasterUwU commented:
@Daru-san Yeah i wanted this in a seperate PR anyways. So just open one when this one gets merged and ping me to let me know.

av-gal reviewed (commented):
You could consider adding an update script like `nix-update`.

```
  passthru.updateScript = nix-update-script {
    extraArgs = [
      "--subpackage"
      "webui"
    ];
  };```

av-gal commented:
You could consider adding an update script like `nix-update`.

```
  passthru.updateScript = nix-update-script {
    extraArgs = [
      "--subpackage"
      "webui"
    ];
  };```

gdamjan commented:
please update:
```
hash = "sha256-Meztr/UxLgnbd3YwkSW0vy+D2N4mFg2v+T4nBnYiQBI=";
```
and
```
npmDepsHash = "sha256-vib8jpf7Jn1qv0m/dWJ4TbisByczNbtEd8hIM5ll2Q8=";
```

ToasterUwU commented:
Yes im aware, just busy atm. Give me some time

av-gal commented:
I'm having a go at packaging rqbit-desktop. It's complicated... the desktop UI reuses code from the web ui, and the typescript compiler doesn't know how to read the cache generated by npmDeps. So you need to create a node_modules for the webui and desktop. This doesn't interact well with the hooks for npm, since they rewrite `package_lock.json` to refer to the source code in the Nix store. So I think you might have to create a derivation from the source, with the two `node_modules` in the right places, and run `buildRustPackage` against that for the build to function correctly.

Haven't figured it out yet though. I'll make a PR if I do.

ToasterUwU commented:
I updated the remaining hashes, i removed the darwin line and the import of the darwin namespace, i added the update script snippet that @av-gal suggested.

Im around more now, so feel free to suggest changes, i will get to them within a day

FliegendeWurst commented on pkgs/by-name/rq/rqbit/package.nix:
```diff
@@ -45,15 +45,11 @@ in
   inherit pname version src;
 
-  cargoHash = "sha256-esDUzzVm5J8fKftBfk5StJzN1YzLa1p0t7BsoxzrowI=";
+  cargoHash = "sha256-cJRv+b780daYeiAFnbhYoYNwOy9GL8x9Ve6XxG95QpU=";
```
```suggestion
  useFetchCargoVendor = true;
  cargoHash = "";
```
We are moving to our new cargo fetcher nixpkgs-wide, please regenerate the hash.
LGTM otherwise, though I did not yet test it.

ToasterUwU committed:
rqbit: 7.0.1 -> 8.0.0
